runtime/integration-tests/src/xcm_mock/xcm_tests.rs

```diff
@@ -3605,3 +3605,50 @@ fn send_disabled_asset_should_fail() {
         ),)
     });
 }
+
+#[test]
+fn transfer_reserve_asset_instruction_from_parachains_is_blocked() {
+    MockNet::reset();
+
+    ParaA::execute_with(|| {
+        let dummy_asset = MultiAsset {
+            id: Concrete(MultiLocation {
+                parents: 1,
+                interior: X1(Parachain(PARA_B_ID)),
+            }),
+            fun: Fungible(1000000000),
+        };
+        let dummy_assets = MultiAssets::from(vec![dummy_asset.clone()]);
+        // Send withdraw and deposit
+        assert_ok!(ParachainPalletXcm::send_xcm(
+            Here,
+            (Parent, Parachain(PARA_B_ID)),
+            Xcm(vec![
+                WithdrawAsset(dummy_assets),
+                BuyExecution {
+                    fees: dummy_asset.clone(),
+                    weight_limit: Unlimited,
```

Comment:
Can you cover `Limited` weight?

Reply:
Or this pr is fine me

Reply:
yeah it doesn't make a difference here, the thing we care about is that the TransferReserveAsset instruction never gets executed.